project/src/components/city-card/city-card.tsx

```diff
@@ -1,35 +1,44 @@
 import { Offer } from '../../types/Offer';
-import { useState } from 'react';
+import BookmarkButton from '../bookmark-button/bookmark-button';
+import CardMark from '../card-mark/card-mark';
+import { getRating, capitalizeFirstLetter } from '../../utils';
+import cn from 'classnames';
+import { Link } from 'react-router-dom';
 
 type CityCardProps = {
   className: string;
+  classNameWrapper: string;
   offer: Offer;
-  isInRoom: boolean;
+  onActiveCardId?: (id: number | null) => void;
 }
 
-const CityCard = ({ className, offer, isInRoom}: CityCardProps): JSX.Element => {
-  const { previewImage, isPremium, price, isFavorite, title, type, } = offer;
-  const [opacity, setOpacity] = useState(false);
-  const onCityCardMouseOver = (): void => {
-    setOpacity(!opacity);
-    console.log(offer.id);
+const CityCard = ({ className, classNameWrapper, offer, onActiveCardId }: CityCardProps): JSX.Element => {
+  const { previewImage, isPremium, price, isFavorite, title, type, rating, id} = offer;
+
+  const handleCityCardOnMouseOver = (cityCardId: number): void => {
+    onActiveCardId && onActiveCardId(cityCardId);
```

Comment:
и можно не выносить отдельно
```suggestion
    onActiveCardId?.(cityCardId);
```